cl/search/views.py

```diff
@@ -52,8 +53,22 @@ def get_solr_result_objects(cd, facet):
     """
     search_type = cd['type']
     if search_type == 'o':
-        si = ExtraSolrInterface(settings.SOLR_OPINION_URL, mode='r')
-        results = si.query().add_extra(**build_main_query(cd, facet=facet))
+        # This is a `related:<pk>` prefix query?
+        related_prefix_match = re.search(r'(^|\s)(?P<pfx>related:(?P<pk>[0-9]+))($|\s)', cd['q'])
+        if related_prefix_match:
+            results = get_mlt_query(
```

Comment:
Should this better parallel the API on lines 69-70? That is, rather than passing all of these parameters, and getting back results, could we pass fewer parameters, and have a `build_mlt_query` function instead? I'm not saying which is better, but consistency is good.

Reply:
I don't see how we can make this better currently since for MLT we need another Solr interface (`MoreLikeThisHighlightsSolrInterface`).